src/common/src/array/mod.rs

```diff
@@ -156,6 +156,11 @@ pub trait Array: std::fmt::Debug + Send + Sync + Sized + 'static + Into<ArrayImp
     /// Retrieve a reference to value.
     fn value_at(&self, idx: usize) -> Option<Self::RefItem<'_>>;
 
+    /// # Safety
+    ///
+    /// Retrieve a reference to value without checking the index boundary.
+    unsafe fn value_at_unchecked(&self, idx: usize) -> Option<Self::RefItem<'_>>;
```

Comment:
I would prefer to implement it as `Index` trait so that we can use `arr[idx]` syntax.

Reply:
Just ignore this comment. It’s impossible to meet its signatures. I’ve tried and failed on it :(